src/OpenTelemetry/Metrics/MetricType.cs

```diff
@@ -7,7 +7,11 @@ namespace OpenTelemetry.Metrics;
 /// Enumeration used to define the type of a <see cref="Metric"/>.
 /// </summary>
 [Flags]
+#pragma warning disable CA1028 // Enum storage should be Int32
+#pragma warning disable CA2217 // Do not mark enums with FlagsAttribute
 public enum MetricType : byte
+#pragma warning restore CA2217 // Do not mark enums with FlagsAttribute
```

Comment:
Something seems off to me about the CA2217.
It should be fine for an Enum to be a Flag, which suggests that something here may not be implemented properly.

Taking a closer look, is it appropriate for MetricType to also be a Flag? For example, is there a scenario where a metric would be both a `Sum` and a `SumNonMonotonic` ?

NOTE TO SELF: should review the spec requirements and the tests for this.

Reply:
I doubt that it should be used as a Flags in general:

Lets consider only this 4 values
hex - bin - dec - enum name
0x1a - 00011010 - 26 - LongSum
0x1d - 00011101 - 29 - DoubleSum
0x2a - 00101010 - 42 - LongGauge
0x2d - 00101101 - 45 - DoubleGauge

Now, lets do:
LongSum | DoubleGauge = 00011010 | 00101101 = 00111111.

It means that also DoubleSum and LongGauge are in scope this combination. It leads to CA2217.

I do not know if we can do with this, without 2.0 release.

Reply:
If not now, should we change it later? Maybe we should create an issue, or things like this would easily get lost.